src/solvers/dg_multi/flux_differencing.jl

```diff
@@ -0,0 +1,200 @@
+"""
+    hadamard_sum_A_transposed!(du, ATr, volume_flux, u, skip_index=(i,j)->false)
+
+Computes the flux difference ∑_j A[i, j] * f(u_i, u_j) and accumulates the result into `du`.
+- `du`, `u` are vectors
+- `ATr` is the transpose of the flux differencing matrix `A`. The transpose is used for
+  faster traversal since matrices are column major in Julia.
+"""
+@inline function hadamard_sum_A_transposed!(du, ATr, volume_flux, u, skip_index=(i,j)->false)
+  rows, cols = axes(ATr)
+  for i in cols
+    u_i = u[i]
+    du_i = du[i]
+    for j in rows
+      if !skip_index(i,j)
+          du_i += ATr[j,i] * volume_flux(u_i, u[j])
```

Comment:
If you wrap this file using the same `@muladd` blokc (including the comment) and change this line to
```suggestion
          du_i = du_i + ATr[j,i] * volume_flux(u_i, u[j])
```
you will get some speed-up by using FMAs (`@muladd` cannot handle `+=` at the moment).